pkg/table/constraint.go

```diff
@@ -51,19 +51,24 @@ func LoadCheckConstraint(tblInfo *model.TableInfo) ([]*Constraint, error) {
 func removeInvalidCheckConstraintsInfo(tblInfo *model.TableInfo) {
 	// check if all the columns referenced by check constraint exist
 	var conInfos []*model.ConstraintInfo
+	var hasChanged bool
 	for _, cons := range tblInfo.Constraints {
 		valid := true
 		for _, col := range cons.ConstraintCols {
 			if tblInfo.FindPublicColumnByName(col.L) == nil {
 				valid = false
+				hasChanged = true
 				break
 			}
 		}
 		if valid {
 			conInfos = append(conInfos, cons)
 		}
 	}
-	tblInfo.Constraints = conInfos
+	// only update the constraints if there are invalid constraints
+	if hasChanged {
+		tblInfo.Constraints = conInfos
+	}
```

Comment:
Why change here？

Reply:
LoadTableInfos may has data race in ut. Because table.FromMetas may read and change constraints here

Reply:
This change does not seem to solve the data race issue.